File: LibraryMe.API/BookLibrary.DAL/Models/DTO/AuthorDTO.cs
namespace BookLibrary.DAL.Models.DTO
{
    public class AuthorDTO
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Patronymic { get; set; }
        public string Biography { get; set; }
        public DateOnly DateOfBirth { get; set; }
        public DateOnly? DateOfDeath { get; set; }
        public string ImageUrl { get; set; }
        public IEnumerable<BookShortcutDTO> Books { get; set; }
    }
}

File: LibraryMe.UI/LibraryMe/src/app/Features/Authors/models/author-dto.ts
import {BookShortcutDTO} from "../../Books/models/book-shortcut-dto";

export interface AuthorDTO {
  name: string;
  surname: string;
  patronymic: string;
  biography: string;
  dateOfBirth: string; // Assuming DateOnly is represented as a string, adjust as needed
  dateOfDeath?: string | null;
  imageUrl: string;
  books: BookShortcutDTO[];
}
